### DISCUSSION TITLE
Controlling the DPI of images generated by Neper

### DISCUSSION THREAD

**Discussion by danishTUE** [neperfepx/neper#749] [Q&A] [answered]
Hi Romain,

I am interested in generating .png images of tesselations with 300 dpi using Neper -V module. How can I control the dpi of the images generated? I see that one can control the total number of pixels in an image but couldn't find anything to control the pixel density. Kindly let me know if there is a possibility to do so.

Regards,

Danish

**Comment by rquey** [comment, marked as answer]
The resolution in dpi is proportional to the number of pixels (along a dimension of the image) and inversely proportional to the image length / width.  Typically, from the desired length / width and dpi resolution, you would compute the necessary number of pixels.

The dpi resolution is not encoded in the png file.

**Reply by danishTUE** [reply]
Thanks Romain.

I do not understand this completely. 

Suppose I want to create an image of 300 dpi, let say in L inch x L inch of the image size. I would calculate the number of pixels needed to make it, which would be 300Lx300L. If I now make Neper -V to create a .png image with 300Lx300L pixels, it will do so by making an image of a certain size (ley say lxl inch). In that case the net DPI would be 300 L/l, which may not be equal to 300.

Am I missing something here. It would be very helpful if you can explain. Thanks!

**Reply by rquey** [reply]
Image of 3 inches x 2 inches, 300 dots per inch -> (300 x 3) x (300 x 2) = 900 x 600 dots (or pixels).  This is your image size.

**Reply by danishTUE** [reply]
900 x 600 is the image size in pixels, which is generated by Neper. But this Neper-generated png image would be of a certain size (in physical dimensions), which would not be 3 inches x 2 inches.

**Reply by rquey** [reply]
Again, the dpi resolution (or the size (physical dimension)) is not encoded in the png file.  It is up to you to choose the image size.

http://news.povray.org/povray.newusers/thread/%3C548a422d%241%40news.povray.org%3E/
